framework/src/kernels/ADDiffusion.C

```diff
@@ -17,6 +17,16 @@ defineADValidParams(
     params.addClassDescription("Same as `Diffusion` in terms of physics/residual, but the Jacobian "
                                "is computed using forward automatic differentiation"););
 
+template <ComputeStage T>
```

Comment:
Do you have to maintain the existing valid parameters for the classes you convert over?

Reply:
NVM - I didn't see your comment before I wrote this.

Reply:
Yes - until all child classes that depend on the old functions have been updated as well.

Reply:
Should we attempt to call one of these functions from the other where we can to avoid maintaining two separate functions?